docs/architecture.md

```diff
@@ -0,0 +1,96 @@
+# RPM Build Pipeline Architecture
+
+In short, the RPM Build Pipeline builds on top of existing tooling that Fedora,
+CentOS, and RHEL maintainers are familiar with—namely, it obtains source code
+from [DistGit][] and builds using [Mock][].
+
+This brings us a lot of convenience, because people can (continue to) work with
+package Git repositories the way they are used to—and, for example, easily
+reproduce builds locally.
+
+People often ask why we use Mock in a "container-native" build system.  The
+short answer is that Mock actually brings *a lot* of convenience.  And note
+this: **Mock is a container-native tool** nowadays!  For more info, check the
+[Why Mock][] document.
+
+
+## Basic pipeline flow
+
+The chart below illustrates the current Tekton Task flow used in the RPM Build
+Pipeline.
+
+```mermaid
+graph TD
+
+  clone_repository[clone-repository] --> get_sources[get-rpm-sources /from DistGit lookaside/]
+
+  get_sources --> nosrpm_a[calculate-deps-x86_64]:::ARCH
+  get_sources --> nosrpm_b[calculate-deps-aarch64]:::ARCH
+  get_sources --> nosrpm_c[calculate-deps ...]:::ARCH
+
+  nosrpm_a --> prefetch[Download BuildRequires.
+    Pull Mock bootstrap.
+    Prepare offline repositories.]
+  prefetch --> build_a[rpmbuild-x86_64]:::ARCH
+
+  nosrpm_b --> prefetch
+  prefetch --> build_b[rpmbuild-aarch64]:::ARCH
+
+  nosrpm_c --> prefetch
+  prefetch --> build_c[rpmbuild ...]:::ARCH
+
+  build_a --> Upload[upload-to-quay]
+  build_b --> Upload
+  build_c --> Upload
+
+  Upload --> check_noarch[check-noarch]
+
+  classDef ARCH fill:yellow
+```
+
+The yellow boxes represent architecture-specific tasks.  This means that, for
+example, `rpmbuild-x86_64` Task must be executed in a native `x86_64`
+environment.  Konflux needs to allocate a virtual machine for this task
+using [MPC][].
+
+
+## Steps
+
+- **clone-repository**
+    - This step is architecture-agnostic and is executed only once.
+    - Typically, we clone the package source from a DistGit repository or a fork.
+    - A full clone is required (to keep [rpmautospec][] happy).
+- **get-rpm-sources**
+    - Downloads source artifacts (e.g., source tarballs) from the corresponding
+      DistGit instance.
+- **calculate-deps-&lt;ARCH&gt; (Mock)**
+    - This step **is not** [hermetic][].
```

Comment:
It would worth to mention that this step enables hermetic rpmbuild process that happens further down the pipeline. To provide a context for reader why this step exists.